packages/models-library/tests/test_basic_regex.py

```diff
@@ -3,74 +3,96 @@
 # pylint:disable=redefined-outer-name
 # pylint:disable=protected-access
 import re
+from datetime import datetime
+from typing import Optional, Sequence
 
+import pytest
+from models_library.basic_regex import DATE_RE, UUID_RE, VERSION_RE
 from pkg_resources import parse_version
 
-from models_library.basic_regex import (
-    PEP404_VERSION_RE,
-    PEP440_VERSION_NG_RE,
-    SEMANTIC_VERSION_NG_RE,
+INVALID = object()
+VALID = object()
+NOT_CAPTURED = ()
+
+
+def assert_match_and_get_capture(regex_str, test_str, expected) -> Optional[Sequence]:
+    match = re.match(regex_str, test_str)
+    if expected is INVALID:
+        assert match is None
+    elif expected is VALID:
+        assert match is not None
+        print(regex_str, "captured:", match.group(), "->", match.groups())
+    else:
+        captured = match.groups()
+        assert captured == expected
+        return captured
+    return None
+
+
+@pytest.mark.parametrize(
+    "version_str, expected",
+    [
+        ("0.10.9", ("0", ".9", "9", None, None, None, None, None, None)),
+        ("01.10.9", INVALID),
+        ("2.1.0-rc2", ("2", ".0", "0", "-rc2", "rc2", None, None, None, None)),
```

Comment:
you are not using this to validate the version right? this will fail on the platform.

Reply:
this is a preparation to understand better how the version REs work and how the captures are taken. It will help us preventing some accidental changes in the regex strings